src/view/test2.java

```diff
@@ -0,0 +1,60 @@
+package view;
```

Comment:
I ran your test suite and got the following errors:

<img width="1335" alt="Screenshot 2023-12-02 at 11 42 09 AM" src="https://github.com/JasonBarahan/TrackMyTransit/assets/80921817/9fb03790-7913-49d4-bf62-21dba5e35a11">

You have some list indexing issues, where I suspect your test ArrayList does not match the format of the ArrayList you would parse from the API call. 

Running the program and tracing output statements, the ArrayList from the API call you then "parse the data of" in `VisualizeInteractor.java` has a form like:

<img width="1331" alt="Screenshot 2023-12-02 at 11 52 56 AM" src="https://github.com/JasonBarahan/TrackMyTransit/assets/80921817/0eb35b91-2e2b-445d-96c7-5081a79990db">

whereas in your test case:

<img width="418" alt="Screenshot 2023-12-02 at 11 54 25 AM" src="https://github.com/JasonBarahan/TrackMyTransit/assets/80921817/9fa3c57e-e208-402e-8f76-404efae40135">

I believe you just "add the values" in without the descriptive tags (ie: You do not do "Vehicle Latitude": -43.0712 for example)


Try making your test ArrayList formatted like this ^^, so that when you call the `substring()` methods, there is no longer a indexing issue

Reply:
`test2.java` was created before I knew how the returned data would be structured (I had inserted a mock implementation into the `Train.java` file) as to not affect the development timeline. By the time the Show Incoming Vehicles use case #37 was implemented (which included the `List<List<String>>` of incoming vehicles) I had updated all the other use case files accordingly, but not this one.

This has since been fixed; refer to #60